ocis-reva/pkg/flagset/storagemetadata.go

```diff
@@ -0,0 +1,79 @@
+
+import (
+	"github.com/micro/cli/v2"
+	"github.com/owncloud/ocis/ocis-reva/pkg/config"
+	"path"
+)
+
+// StorageMetadata applies cfg to the root flagset
+func StorageMetadata(cfg *config.Config) []cli.Flag {
+	flags := []cli.Flag{
+		&cli.StringFlag{
+			Name:        "debug-addr",
+			Value:       "0.0.0.0:9184",
+			Usage:       "Address to bind debug server",
+			EnvVars:     []string{"REVA_STORAGE_METADATA_DEBUG_ADDR"},
+			Destination: &cfg.Reva.StorageMetadata.DebugAddr,
+		},
+		&cli.StringFlag{
+			Name:        "network",
+			Value:       "tcp",
+			Usage:       "Network to use for the reva service, can be 'tcp', 'udp' or 'unix'",
+			EnvVars:     []string{"REVA_STORAGE_METADATA_NETWORK"},
+			Destination: &cfg.Reva.StorageMetadata.Network,
+		},
+		&cli.StringFlag{
+			Name:        "provider-addr",
+			Value:       "0.0.0.0:9185",
+			Usage:       "Address to bind reva service",
+			EnvVars:     []string{"REVA_STORAGE_METADATA_PROVIDER_ADDR"},
+			Destination: &cfg.Reva.StorageMetadata.Addr,
+		},
+		&cli.StringFlag{
+			Name:        "data-server-url",
+			Value:       "http://localhost:9187",
+			Usage:       "URL of the data-server the storage-provider uses",
+			EnvVars:     []string{"REVA_STORAGE_METADATA_DATA_SERVER_URL"},
+			Destination: &cfg.Reva.StorageMetadata.DataServerURL,
+		},
+		&cli.StringFlag{
+			Name:        "data-server-addr",
+			Value:       "0.0.0.0:9187",
+			Usage:       "Address to bind the metadata data-server to",
+			EnvVars:     []string{"REVA_STORAGE_METADATA_DATA_SERVER_ADDR"},
+			Destination: &cfg.Reva.StorageMetadataData.Addr,
+		},
+		&cli.StringFlag{
+			Name:        "storage-provider-driver",
+			Value:       "local",
+			Usage:       "storage driver for metadata mount: eg. local, eos, owncloud, ocis or s3",
+			EnvVars:     []string{"REVA_STORAGE_METADATA_PROVIDER_DRIVER"},
+			Destination: &cfg.Reva.StorageMetadata.Driver,
+		},
+		&cli.StringFlag{
+			Name:        "data-provider-driver",
+			Value:       "local",
+			Usage:       "storage driver for data-provider mount: eg. local, eos, owncloud, ocis or s3",
+			EnvVars:     []string{"REVA_STORAGE_METADATA_DATA_PROVIDER_DRIVER"},
+			Destination: &cfg.Reva.StorageMetadataData.Driver,
+		},
+	}
+
+	flags = append(flags, TracingWithConfig(cfg)...)
+	flags = append(flags, DebugWithConfig(cfg)...)
+	flags = append(flags, SecretWithConfig(cfg)...)
+	flags = append(flags, DriverEOSWithConfig(cfg)...)
+	flags = append(flags, DriverLocalWithConfig(cfg)...)
+	flags = append(flags, DriverOwnCloudWithConfig(cfg)...)
+	flags = append(flags, DriverOCISWithConfig(cfg)...)
+
+	// Metadata storage needs its own root
+	cfg.Reva.Storages.Common.Root = path.Join(cfg.Reva.Storages.Common.Root, "metadata")
+	cfg.Reva.Storages.OwnCloud.Root = path.Join(cfg.Reva.Storages.OwnCloud.Root, "metadata")
+	cfg.Reva.Storages.EOS.Root = path.Join(cfg.Reva.Storages.EOS.Root, "metadata")
+	cfg.Reva.Storages.Local.Root = path.Join(cfg.Reva.Storages.Local.Root, "metadata")
```

Comment:
this appends `metadata` to the default path, which is
- Reva.Storages.Common.Root: `/var/tmp/ocis/root` for ocis
- Reva.Storages.EOS.Root: `/eos/dockertest/reva`
- Reva.Storages.Local.Root: `/var/tmp/reva/root`
- Reva.Storages.OwnCloud.Root: `/var/tmp/reva/data`

ultimately, I would like to use the common config for as many drivers as possible. Anyway, the root for metadata might be very different from the root for user data, and making it a subfolder might even conflict with other storage providers.

To really give the metadata driver its own root we need to parse the flag:
```
		&cli.StringFlag{
			Name:        "storage-metadata-root",
			Value:       "/var/tmp/ocis/metadata",
			Usage:       "the path to the metadata storage root",
			EnvVars:     []string{"REVA_STORAGE_METADATA_ROOT"},
			Destination: &cfg.Reva.Storages.Common.Root,
		},
```

This would then be used as the EOS driver root, the local driver root, the owncloud driver root or the ocis driver root. 

I would use the `reva-storage-metadata` command as a blueprint for a `reva-storage-home` and `reva-storage-users` command, that would replace the existing commands and to combine them with the separate data provider commands. but that is a subsequent PR.